### PR TITLE
Fix negative axis issue with ragged tensor and reduce_sum

### LINKED ISSUE #27497: `tf.reduce_sum` with multiple negative axes and `tf.RaggedTensor` bugged
## System information
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): ubuntu 16.04
- TensorFlow installed from (source or binary): pip
- TensorFlow version (use command below):
- Python version: ('v1.13.1-0-g6612da8951', '1.13.1')
- GPU model and memory: quadro k620 2gb

## Current behaviour
Multiple axes passed to `tf.reduce_sum` with first argument being a ragged tensor results in incorrect behaviour.

```python
import tensorflow as tf

x_values = tf.random.normal(shape=(100, 5, 6))
x_row_lengths = tf.constant([20, 30, 50], dtype=tf.int64)
x_ragged = tf.RaggedTensor.from_row_lengths(x_values, x_row_lengths)
print(x_ragged.shape)
# [3, ?, 5, 6]

# wrong shape
print(tf.reduce_sum(x_ragged, axis=(-2, -3)).shape)
# [50, 6]

# positive axes work
print(tf.reduce_sum(x_ragged, axis=(1, 2)).shape)
# [3, 6]

# separate reductions work
print(tf.reduce_sum(tf.reduce_sum(x_ragged, axis=-3), axis=-2).shape)
# [3, 6]
```

## Expected behaviour
Same result as corresponding positive indices/separate reductions.

### PR DESCRIPTION
**Pull request by yongtang**
This fix tries to address the issue raised in #27497 where `tf.reduce_sum` with multiple negative axes and ragged tensor does not produce correct result.

The issue is that during reduce op, ragged tensor will reduce one axis at a time. However, for negative axis, sort result is reversed so order is different.

This fix convert to positive before the sort to make sure the order.

This fix fixes #27497.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

**Review by edloper — APPROVED** [review]

**Review by alextp — APPROVED** [review]

**Comment by rthadur** [comment]
@yongtang can you please check build failures
